speechbrain/lobes/models/FastSpeech.py

```diff
@@ -0,0 +1,396 @@
+                                        kdim=enc_k_dim,
+                                        vdim=enc_v_dim,
+                                        dropout=enc_dropout,
+                                        activation=nn.ReLU,
+                                        normalize_before=normalize_before,
+                                        ffn_type=ffn_type)
+
+        self.decoder = TransformerEncoder(num_layers=dec_num_layers,
+                                        nhead=dec_num_head,
+                                        d_ffn=dec_ffn_dim,
+                                        d_model=dec_d_model,
+                                        kdim=dec_k_dim,
+                                        vdim=dec_v_dim,
+                                        dropout=dec_dropout,
+                                        activation=nn.ReLU,
+                                        normalize_before=normalize_before,
+                                        ffn_type=ffn_type)
+
+        self.linear = nn.Linear(dec_d_model, n_mels)
+
+
+    def forward(self, tokens, durations=None):
+        """forward pass for training and inference
+        Arguments
+        ---------
+        tokens: torch.tensor
+            batch of input tokens
+        durations: torch.tensor
+            batch of durations for each token. If it is None, the model will infer on predicted durations
+        Returns
+        ---------
+        mel_post: torch.Tensor
+            mel outputs from the decoder
+        predict_durations: torch.Tensor
+            predicted durations for each token
+        """
+        token_feats = self.encPreNet(tokens)
+        srcmask = get_key_padding_mask(tokens, pad_idx=self.padding_idx)
+        attn_mask = srcmask.unsqueeze(-1).repeat(self.enc_num_head, 1, token_feats.shape[1])
+        token_feats, memory = self.encoder(token_feats, src_mask=None, src_key_padding_mask=srcmask)
+
+        predict_durations = self.durPred(token_feats).squeeze()
+        if predict_durations.dim() == 1: predict_durations = predict_durations.unsqueeze(0)
+        if durations is None: dur_pred_reverse_log = torch.clamp(torch.exp(predict_durations) - 1, 0)
+        spec_feats = upsample(token_feats, durations if durations is not None else dur_pred_reverse_log)
+
+        srcmask = get_key_padding_mask(spec_feats, pad_idx=self.padding_idx)
+        attn_mask = srcmask.unsqueeze(-1).repeat(self.dec_num_head, 1, spec_feats.shape[1])
+        output_mel_feats, memory, *_ = self.decoder(spec_feats, src_mask=None, src_key_padding_mask=srcmask)
+
+        mel_post = self.linear(output_mel_feats)
+        return mel_post, predict_durations
+
+def upsample(feats, durs, pace=1, padding_value=0.0):
+    """upsample encoder ouput according to durations
+    Arguments
+    ---------
+    feats: torch.tensor
+        batch of input tokens
+    durs: torch.tensor
+        durations to be used to upsample
+    pace: int
```

Comment:
`pace` is an important parameter but not used in the upsample function, while changing the speed is one of the best features of Fastspeeh. I think we should use `int(pace * durs[i])` instead of `durs[i]`, and we add it as a hyperparameter in yaml